src/BenchmarkDotNet/Detectors/Cpu/ICpuDetector.cs

```diff
@@ -0,0 +1,12 @@
+using Perfolizer.Phd.Dto;
+
+namespace BenchmarkDotNet.Detectors.Cpu;
+
+/// <summary>
+/// Loads the <see cref="PhdCpu"/> for the current hardware
+/// </summary>
+public interface ICpuDetector
```

Comment:
does this interface need to be public? can it be internal?

Reply:
Eventually, I want to come up with an API design, in which all such features can be easily redefined by the user. Imagine a request like #2577. Right now, if there is a bug in CPU detection and we want to introduce a fix, the verification feedback loop is too long: we should merge the potential fix in master, deploy the nightly version, next ask the user to upgrade their BDN to the latest nightly and check how it works. It would be much more convenient if the user had the ability to play with a potential fix in their local project without a need for updated nightly packages. Also, such a way allows the user to fix the problem on the spot without waiting for the next official BenchmarkDotNet release.

However, I don't have a strict opinion on the exact API. We can change it whatever we want.